File: src/main/java/de/maggiwuerze/xdccloader/model/forms/UserForm.java
package de.maggiwuerze.xdccloader.model.forms;

import de.maggiwuerze.xdccloader.model.forms.validation.PasswordMatches;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.NoArgsConstructor;

@PasswordMatches
@Data
@NoArgsConstructor
public class UserForm {

	@NotNull
	@NotEmpty
	private String username;

	@NotNull
	@NotEmpty
	private String password;

	@NotNull
	@NotEmpty
	private String matchingPassword;
}

File: src/main/kotlin/de/maggiwuerze/xdccwebloader/model/forms/UserForm.kt
package de.maggiwuerze.xdccwebloader.model.forms

import de.maggiwuerze.xdccwebloader.model.forms.validation.PasswordMatches
import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.NotNull

@PasswordMatches


class UserForm {
    @NotNull
    @NotEmpty
    var username: String? = null

    @NotNull
    @NotEmpty
    var password: String? = null

    @NotNull
    @NotEmpty
    var matchingPassword: String? = null
}
